### PR TITLE
chore: cherrypick Renovate updates

### PR DESCRIPTION
**Pull request by hoobdeebla**
Includes all updates from #25546 except for express-graphql, fixes build/test issues from those updates, and adds express-graphql to the <1.0.0-minor group in the Renovate config.

### CODE REVIEW

**Comment by pvdz** [comment]
I appreciate the cherry pick but this is too much to review. It'll have to be merged in smaller chunks or as renovatebot's own PR. Many packages have subtle (and not so subtle) breaking changes which is why we can't blindly upgrade, even minor bumps, and sometimes even patch bumps (although they're usually fine).

I'm not sure what to do with this one but I'm inclined to close it.

Please either take a smaller chunk so we can verify what's being bumped and why that's not breaking, or leave it to renovate...

**Comment by hoobdeebla** [comment]
Thank you for the feedback! I reverted most updates to allow Renovate to do most of the work but kept those that required snapshot updates (katex) or code changes (@pmmmwh/react-refresh-webpack-plugin). I also kept the Renovate config change that adds express-graphql to the <1.0.0-minor group to avoid merging breaking changes from that module in the future. Finally, I cherrypicked updates for chokidar, webpack, whatwg-fetch, and a few dependencies of gatsby-telemetry.

Here is a table of the included updates:

Package | Type | Update | Change
-- | -- | -- | --
[@pmmmwh/react-refresh-webpack-plugin](https://togithub.com/pmmmwh/react-refresh-webpack-plugin) | dependencies | minor | [`^0.3.3` -> `^0.4.1`](https://renovatebot.com/diffs/npm/@pmmmwh%2freact-refresh-webpack-plugin/0.3.3/0.4.1)
[@pmmmwh/react-refresh-webpack-plugin](https://togithub.com/pmmmwh/react-refresh-webpack-plugin) | devDependencies | minor | [`^0.3.3` -> `^0.4.1`](https://renovatebot.com/diffs/npm/@pmmmwh%2freact-refresh-webpack-plugin/0.3.3/0.4.1)
[chokidar](https://togithub.com/paulmillr/chokidar) | dependencies | patch | [`3.4.0` -> `^3.4.2`](https://renovatebot.com/diffs/npm/chokidar/3.4.0/3.4.2)
[chokidar](https://togithub.com/paulmillr/chokidar) | devDependencies | patch | [`^3.4.0` -> `^3.4.2`](https://renovatebot.com/diffs/npm/chokidar/3.4.0/3.4.2)
[envinfo](https://togithub.com/tabrindle/envinfo) | dependencies | minor | [`^7.5.1` -> `^7.7.3`](https://renovatebot.com/diffs/npm/envinfo/7.5.1/7.7.3)
[eslint-plugin-react](https://togithub.com/yannickcr/eslint-plugin-react) | dependencies | patch | [`^7.20.2` -> `^7.20.6`](https://renovatebot.com/diffs/npm/eslint-plugin-react/7.20.2/7.20.6)
[eslint-plugin-react](https://togithub.com/yannickcr/eslint-plugin-react) | devDependencies | patch | [`^7.20.2` -> `^7.20.6`](https://renovatebot.com/diffs/npm/eslint-plugin-react/7.20.2/7.20.6)
[git-up](https://togithub.com/IonicaBizau/git-up) | dependencies | patch | [`4.0.1` -> `^4.0.2`](https://renovatebot.com/diffs/npm/git-up/4.0.1/4.0.2)
[is-docker](https://togithub.com/sindresorhus/is-docker) | dependencies | minor | [`2.0.0` -> `^2.1.1`](https://renovatebot.com/diffs/npm/is-docker/2.0.0/2.1.1)
[katex](https://katex.org/) ([source](https://togithub.com/KaTeX/KaTeX)) | devDependencies | minor | [`^0.11.1` -> `^0.12.0`](https://renovatebot.com/diffs/npm/katex/0.11.1/0.12.0)
[webpack](https://togithub.com/webpack/webpack) | dependencies | minor | [`~4.43.0` -> `^4.44.1`](https://renovatebot.com/diffs/npm/webpack/4.43.0/4.44.1)
[webpack](https://togithub.com/webpack/webpack) | devDependencies | minor | [`^4.43.0` -> `^4.44.1`](https://renovatebot.com/diffs/npm/webpack/4.43.0/4.44.1)
[whatwg-fetch](https://togithub.com/github/fetch) | devDependencies | minor | [`3.0.0` -> `^3.4.0`](https://renovatebot.com/diffs/npm/whatwg-fetch/3.0.0/3.4.0)
[whatwg-fetch](https://togithub.com/github/fetch) | devDependencies | minor | [`^3.1.0` -> `^3.4.0`](https://renovatebot.com/diffs/npm/whatwg-fetch/3.1.0/3.4.0)

**Review by pvdz — CHANGES_REQUESTED** [review]
- [react-refresh](https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/main/CHANGELOG.md) changes look ok
- [chokidar](https://github.com/paulmillr/chokidar/blob/master/.github/full_changelog.md) changes look ok
- [envinfo](https://github.com/tabrindle/envinfo/commits/main) changes look benign, hard to tell, I think it's ok?
- [eslint](https://github.com/yannickcr/eslint-plugin-react/blob/master/CHANGELOG.md) changes look ok
- [git-up](https://github.com/IonicaBizau/git-up/commits/master) seems to be a docs bump
- [is-docker](https://github.com/sindresorhus/is-docker/commits/master) cli seems to have changed the exit code? I think this - doesn't affect us.
- ! [katex](https://github.com/KaTeX/KaTeX/blob/master/CHANGELOG.md) explicitly says they're no longer supporting IE10, so we need to postpone this bump.
- [webpack](https://github.com/webpack/webpack/releases): seems to be ok
- [fetch](https://github.com/github/fetch/commits/master) appears to be good to go.

I think we won't want to merge the katex bump for now due to IE10 so please drop that. After that we can merge this on green.

Thank you :)

File: packages/gatsby-remark-katex/src/__tests__/__snapshots__/index.js.snap
Comment (hoobdeebla) [reply packages/gatsby-remark-katex/src/__tests__/__snapshots__/index.js.snap:35]: Since katex is a devDependency, only the internal development/snapshot output is affected by the bump from `0.11` to `0.12`. The katex peerDependency remains at `^0.10.0` which still has IE9/10 support. AFAIK the plugin still properly renders katex from `0.10` and `0.11` - if the bump was to a direct/peer dependency I would obviously go through with the requested change, but I thought I'd double check this. Would you still like me to revert?

**Comment by pvdz** [comment]
Thank you :purple_heart:

**Pull request merged**
